enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,213 @@
+---
+title: external-lb-vips
+authors:
+  - "@emilienm"
+reviewers: 
+  - "@cybertron for SME review on on-premise networking"
+  - "@celebdor for SME review on on-premise networking"
+  - "@jcpowermac for SME review on installer"
+  - "@patrickdillon for SME review and approval on installer"
+  - "@JoelSpeed for SME review on API and general feedback"
+  - "@deads2k for SME review on API and general feedback"
+approvers:
+  - "@cybertron"
+  - "@patrickdillon"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-01-23
+last-updated: 2023-01-23
+tracking-link:
+  - https://issues.redhat.com/browse/OCPBU-156
+see-also:
+  - "/enhancements/on-prem-service-loadbalancers.md"
+replaces:
+  - https://github.com/openshift/enhancements/pull/1281
+---
+
+# External loadbalancer VIPs
+
+## Summary
+
+Originally the on-premise IPI architecture was designed to deploy an internal
+loadbalancer based on HAproxy, Keepalived and CoreDNS.
+When customers needed more flexibility for the VIPs management, they would
+deploy with the UPI method.
+
+This design is proposing the support for using an external loadbalancer
+for the API and Ingress VIPs, so this can be managed outside of the cluster,
+by the customer.
+
+## Motivation
+
+Allow to configure compute and control plane nodes on across multiple subnets for on-premise IPI deployments.
+With separating nodes in subnets, also allow using an external loadbalancer, instead of the built-in that the
+IPI workflow installs, so that the customer have the flexility to configure their own loadbalancer with the ingress and API VIPs pointing to nodes in the separate subnets and manage how the VIPs are routed how they want (e.g. routed via BGP).
+
+### User Stories
+
+As a deployer of a IPI OpenShift cluster, I want to place my control plane nodes and compute nodes on 3 or
+more subnets at installation time. Therefore I need to be able to configure an external load balancer for
+the VIPs.
+
+### Goals
+
+Allow to replace the internal loadbalancer by an external one and support this scenario when the customer
+know what they're doing.
+
+### Non-Goals
+
+* Making this scenario by default at some point. The internal loadbalancer for on-premise platforms will
+  remain the HAproxy/Keepalived/CoreDNS stack.
+* Supporting a mix of both (internal + external loadbalancers for the API & Ingress VIPs), customers
+  will have to choose a scenario.
+
+## Proposal
+
+We will make some change in `install-config.yaml` so customer can configure the
+deployment to use an external loadbalancer if they want.
+Then the components which interact with networking will have to check this
+new field and handle this scenario appropriately.
```

Comment:
Is there a list of these components and the changes required enumerated somewhere? API server internal URL is set in the infrastructure status right? I assume anything that needs to work with this sources it from there?

Edit: Reading further I see in the workflow that we aren't going to read from here, so where are components like MCO going to find the right addresses?

Reply:
We are going to read the VIPs from PlatformStatus, as it was decided. Let me know if this comment still stands.
Thanks